0001-typedb/02-dev/06-infer.md

```diff
@@ -0,0 +1,336 @@
+```
+
+## Example
+
+In the IAM database there are multiple rules, but we will use the one described at the 
+[IAM schema explanation](../01-start/04-iam-schema.md) page. 
+
+<!-- test-ignore -->
+```typeql
+rule add-view-access:
+    when {
+        $modify isa action, has action-name "modify_file";
+        $view isa action, has action-name "view_file";
+        $ac_modify (accessed-object: $obj, valid-action: $modify) isa access;
+        $ac_view (accessed-object: $obj, valid-action: $view) isa access;
+        (permitted-subject: $subj, permitted-access: $ac_modify) isa permission;
+    } then {
+        (permitted-subject: $subj, permitted-access: $ac_view) isa permission;
+    };
+```
+
+By default, the IAM dataset used in the [Quickstart guide](../01-start/03-quickstart.md) sets only permissions for the 
+`modify_file` action. 
+
+But the IAM schema has the rule called `add-view-access` (see above) to infer `view_file` action access from the 
+`modify_file` access. The `view_file` action access permission does not exist in the database but can be inferred by 
+the rule.
+
+To do so, let’s use the following query:
+
+<!-- test-ignore -->
+```typeql
+match
+    $p isa person, has full-name "Kevin Morrison";
+    $o isa object, has path $o-path;
+    $a isa action, has action-name "view_file";
+    $ac(accessed-object: $o, valid-action: $a) isa access;
+    $pe(permitted-subject: $p, permitted-access: $ac) isa permission;
+get $o-path;
+```
+
+The above example should return `path` values for most files if the inference option is **enabled**. Without the 
+inference enabled the query should not return any results (for the default IAM dataset contains no permissions for 
+access with `view_file` action).
+
+### Query explanation
+
+The example is a `get` query with `match` clause and optional `get` clause that does the following:
+
+1. Finds `person` entity (`$p`) with an attribute `full-name` equal to the `Kevin Morrison` string value.
+2. Finds all `object` entities (`$o`) with a `path` attribute (`$o-path`).
+3. Finds `action` entity (`$a`) with `action-name` of `view-file`.
+4. Finds `access` relation (`$ac`) that relates `$o` (as `accessed-object`) to `$a` (as `valid-action`).
+5. Finds all `permission` relations that relate `$p` (as `permitted-subject`) to `$ac` (as `permitted-access`).
+6. Filters the results to receive only the `$o-path` variable that complies with all the statements above.
+
+In short, it returns `path` attributes for every `object` that `person` with a `full-name` of `Kevin Morrison` has 
+permission to access with `action` with the `action-name` of `view_file`. 
+
+## Explain query
+
+There is specific type of query to use with the inferred results if we want them explained by a TypeDB server.
+
+When we get a result of a `get` query that was performed with inference option **enabled** we can use an **explain** 
+query to request additional information on how the exact instances of data were inferred.
```

Comment:
You also need the `explain` option enabled.

Reply:
Fixed in https://github.com/vaticle/docs/pull/676/commits/53fc726782fd7b82e630a7c3ceb6c8608cc6cc41